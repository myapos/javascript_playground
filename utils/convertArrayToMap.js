import splitDate from './splitDate';
import prettyValue from './prettyValue';
import monthInfo from './monthInfo';

const convertArrayToMap = (ar) => {
  let myMap = new Map();

  ar.forEach((raw) => {
    const splitted = splitDate(raw.Date);

    // detect key
    const prettyKey = prettyValue(splitted[1]);

    myMap.set(raw.Date, {
      ...raw,
      filled: typeof raw.filled !== 'undefined' ? raw.filled : false,
      month: monthInfo[prettyKey].month,
      order: monthInfo[prettyKey].order,
    });
  });

  return myMap;
};
export default convertArrayToMap;
