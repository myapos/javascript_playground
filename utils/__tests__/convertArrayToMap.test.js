import convertArrayToMap from '../convertArrayToMap';
import { rawData } from './__data__/rawData';
describe('convertArrayToMap', () => {
  it('will convert array to Map', () => {
    const filledMap = convertArrayToMap(rawData);
    expect(filledMap).toMatchSnapshot();
  });
});
