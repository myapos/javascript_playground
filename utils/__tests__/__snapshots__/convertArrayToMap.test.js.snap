// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertArrayToMap will convert array to Map 1`] = `
Map {
  "14-03-2020" => Object {
    "Date": "14-03-2020",
    "Kgrs": "108",
    "filled": false,
    "month": "March",
    "order": "03",
  },
  "03-05-2020" => Object {
    "Date": "03-05-2020",
    "Kgrs": "104,9",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "12-05-2020" => Object {
    "Date": "12-05-2020",
    "Kgrs": "103,6",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "18-05-2020" => Object {
    "Date": "18-05-2020",
    "Kgrs": "104,7",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "19-05-2020" => Object {
    "Date": "19-05-2020",
    "Kgrs": "104,2",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "20-05-2020" => Object {
    "Date": "20-05-2020",
    "Kgrs": "104,2",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "21-05-2020" => Object {
    "Date": "21-05-2020",
    "Kgrs": "104,2",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "22-05-2020" => Object {
    "Date": "22-05-2020",
    "Kgrs": "103,7",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "23-05-2020" => Object {
    "Date": "23-05-2020",
    "Kgrs": "103,7",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "24-05-2020" => Object {
    "Date": "24-05-2020",
    "Kgrs": "103,2",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "25-05-2020" => Object {
    "Date": "25-05-2020",
    "Kgrs": "102,8",
    "filled": false,
    "month": "May",
    "order": "05",
  },
  "26-05-2020" => Object {
    "Date": "26-05-2020",
    "Kgrs": "103,1",
    "filled": false,
    "month": "May",
    "order": "05",
  },
}
`;
